### ISSUE TITLE
Compiling a standalone project using MTF [solved, with example Cmake file]

### ISSUE THREAD

**Issue by benoitrosa** [abhineet123/MTF#11]
Hi,

I'm trying to compile the example code provided in the Youtube video on the main MTF webpage. 

I first noticed that there is no cmake config file generated to find MTF, so I tried to make my own cmake project. However, when compiling I consistently get the following error : 

`/usr/local/include/mtf/Utilities/inputUtils.h:16:21: fatal error: XVVideo.h: No such file or directory `

The problem is that the whole MTF package has been compiled without XVision. It's not even installed on my system, and yet MTF is compiled and runnning (runMTF with webcam tracking example is working fine). 

Is there something I'm doing wrong ? Here is a gist with my CMakeLists file : https://gist.github.com/benoitrosa/7efb180f2fc0c7cbdb09d0c1ab3436d1

Thanks !

**Comment by benoitrosa** [comment]
OK, I got the thing working with a bit of trial and error. I updated the gist with a working version, hopefully it will be useful to other people as well. 

In the future, it would be nice to generate a CMake config file during the build process, so that we can easily do find_package !

**Comment by abhineet123** [comment]
Thanks for letting me know about this. I don't know enough cmake yet to generate a config file during build process so any help in this regard is greatly appreciated.

For now, the easiest way to compile a new executable using MTF is to add its entry in `Examples/Examples.cmake` after duplicating one of the existing entries (like `runMTF`), changing the target names and adding the new names to `EX_TARGET_NAMES`, `EX_INSTALL_TARGET_NAMES` and `EX_COMBINED_TARGET_NAMES` variables (lines 28-30).

**Comment by benoitrosa** [comment]
Adding a new entry in the Examples folder and in the Examples.cmake file would require rebuilding the whole MTF package everytime, which is probably not the best thing to do ...

I'll look the cmake utility up when I have a bit of time. I know there are automatic configuration packages for doing this.

**Comment by benoitrosa** [comment]
I managed to get a cleaner build by generating a CMake config file. It's not a file usable by the find_package command, but it simplifies a lot building projects anyway. What I did : 

1 - add a mtfConfig.cmake.in file in the MTF/cmake folder (see syntax here : https://gist.github.com/benoitrosa/0cf91a558639b388ad0736ad4af7bb1a )
2 - add those lines at the end of the main CMakeLists file : 
 `if(WIN32)
	configure_file(cmake/mtfConfig.cmake.in "${PROJECT_BINARY_DIR}/mtfConfig.cmake" NEWLINE_STYLE WIN32)
else()
	configure_file(cmake/mtfConfig.cmake.in "${PROJECT_BINARY_DIR}/mtfConfig.cmake" NEWLINE_STYLE UNIX)
endif() `

3 - run the cmake command as normal. It will generate a mtfConfig.cmake file in the build folder with the main defines. 

I did not put any line in the main cmake file to copy this file anywhere, so at this stage one should just copy the generated config file in his/her project tree, and include it. Then, one can use the defined variables. Here is an example of CMakeLists.txt file for a standalone project, using that generated config file : https://gist.github.com/benoitrosa/8a20d0067efd863780b39726fbc1eb5a


I did not do a PR as this is not a nice solution (the best way wouyld be to autogenerate files usable by cmake's find_package). But feel free to incorporate those if you want to, and/or add a few lines in the doc.
